### PR TITLE
[MAINT] update codespell config

### LINKED ISSUE #1467: update codespell pre-commit config to use our "special" dictionary
Super minor but realized that our current pre-commit config does use our extra codespell checks.

### PR DESCRIPTION
**Pull request by Remi-Gau**
closes #1467

- update pre-commit config to use both the default and our pre-set dictionary
- update codespell config to use US english
  - apply to change remaining "britishisms"
- manually fix remaining "file name -> filename"
   -  AFAICT codespell cannot handle flagging 2 words that should be merged and manual search and replace has to be done
- fix a couple of dead links (links to renamed anchors).

### CODE REVIEW

File: .codespell_dict
Comment (Remi-Gau) [inline .codespell_dict:1]: I think that codespell is case sensitive.

File: src/modality-agnostic-files.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ and a guide for using macros can be found at
       "DatasetType": "RECOMMENDED",
       "License": "RECOMMENDED",
       "Authors": "RECOMMENDED",
-      "Acknowledgements": "OPTIONAL",
```
Comment (Remi-Gau) [inline src/modality-agnostic-files.md:32]: OK we are going to have to ignore this change.

File: src/modality-specific-files/magnetic-resonance-imaging-data.md [hunk 2]
```diff
@@ -1233,7 +1233,7 @@ The [`dir-<label>`](../appendices/entities.md#dir) entity is REQUIRED
 for these files.
 This entity MUST be used in addition to
 the REQUIRED `PhaseEncodingDirection` metadata field
-(see [File name structure](../common-principles.md#file-name-structure)).
```
Comment (Remi-Gau) [inline src/modality-specific-files/magnetic-resonance-imaging-data.md:1236]: IS there a way to detect those "dead anchor" links?

Comment (sappelhoff) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md:1236]: Until now I thought the linkchecker would find those.

**Review by sappelhoff — APPROVED** [review]

File: src/modality-specific-files/magnetic-resonance-imaging-data.md (hunk 2 above)
Comment (Remi-Gau) [reply src/modality-specific-files/magnetic-resonance-imaging-data.md:1236]: Given that this does not fail, I think that we may have to use another strategy
https://bids-specification.readthedocs.io/en/latest/common-principles.html#foo-bar

**Review by effigies — APPROVED** [review]

**Pull request merged**